manifests/params.pp

```diff
@@ -34,7 +34,24 @@
   $postrun_command     = undef
   $dns_alt_names       = []
   $use_srv_records     = false
-  $srv_domain          = $::domain
+
+  # On versions < 3.7.5 there was no defined function as such strict variables cannot be used
+  # On verisons < 4.0.0 and >= 3.5.0 the facts hash was optional disabled by default, on version < 3.5.0 the facts hash did not exist
+  # In some edge cases 'domain' fact may be not defined
+  if versioncmp($::puppetversion, '3.7.5') < 0 {
+    $srv_domain = $::domain
+  } elsif versioncmp($::puppetversion, '4.0.0') >= 0 {
```

Comment:
Sorry I should have been a bit more clear, So the domain error provided above is identical to ::puppetmaster as set by theforeman (in my use case I use the module without foreman)

Here are the 2 entries from a single agent run:
```log
/etc/puppetlabs/code/environments/production/modules/puppet/manifests/params.pp:37:26
2016-12-07 13:22:12,471 WARN [qtp1376229403-70] [puppetserver] Puppet Unknown variable: '::puppetmaster'. at /etc/puppetlabs/code/environments/production/modules/puppet/manifests/params.pp:179:32
```

as you will see $::puppetmaster is wrapped in a defined function, unfortunately for $::puppetmaster I am unable to device a solution for that error, but this PR addresses the $::domain fact.